ct-app/core/model/subgraph/providers.py

```
@@ -42,12 +32,10 @@ def _initialize_query(
         if extra_inputs is None:
             extra_inputs = []
 
-        keys, self._sku_query = self._load_query(query_file, extra_inputs)
-
-        if self.default_key is None:
-            self.default_key = keys
+        self._default_key, self._sku_query = self._load_query(
+            query_file, extra_inputs)
 
-    def _load_query(self, path: Union[str, Path], extra_inputs: list[str] = []) -> str:
+    def _load_query(self, path: str | Path, extra_inputs: list[str] = []) -> str:
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)